### ISSUE TITLE
How to implement dropout?

### ISSUE THREAD

**Issue by samuelazran** [Conchylicultor/DeepQA#19]
Hi Everyone!

I was wondering what is the meaning of the warning on this line in model.py? Is it about just adding a condition to add this layer only if test mode? I think it could be interesting to experiment with drop out to see how it can improve generalization. Also what are recommended hyper parameter values to try?

`            #encoDecoCell = tf.nn.rnn_cell.DropoutWrapper(encoDecoCell, input_keep_prob=1.0, output_keep_prob=1.0)  # TODO: Custom values (WARNING: No dropout when testing !!!)
`

Thanks for this great repo!

Have a nice day

**Comment by Conchylicultor** [comment]
It's just a message for my future self to remind me that if this line is uncommented, there will also be dropout when testing (which is bad). The solution should be to either surround this declaration with `if not self.args.test:` (easy) or to control the dropout rate with a placeholder value (better).

I also think it should be interesting to see the effect of dropout on the results. Don't hesitate to share your results if you try.
